app/src/main/java/org/fossasia/susi/ai/chat/adapters/viewholders/TableViewHolder.java

```diff
@@ -100,55 +100,40 @@ public void setView(final ChatMessage model) {
-            setRating(true, true);
-        } else if (model.isPositiveRated() && !model.isNegativeRated()) {
-            setRating(false, true);
-            thumbsUp.setImageResource(R.drawable.thumbs_up_outline);
-            rateSusiSkill(Constant.NEGATIVE, model.getSkillLocation(), itemView.getContext());
-            sleep(500);
-            rateSusiSkill(Constant.NEGATIVE, model.getSkillLocation(), itemView.getContext());
-            setRating(true, false);
-        } else if (!model.isPositiveRated() && model.isNegativeRated()) {
-            rateSusiSkill(Constant.POSITIVE, model.getSkillLocation(), itemView.getContext());
-            setRating(false, false);
-            thumbsDown.setImageResource(R.drawable.thumbs_down_outline);
-        }
-    }
+        thumbsUp.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View view) {
+                Timber.d(model.isPositiveRated() + " " + model.isNegativeRated());
+                if (!model.isPositiveRated() && !model.isNegativeRated()) {
+                    thumbsUp.setImageResource(R.drawable.thumbs_up_solid);
+                    rateSusiSkill(Constant.POSITIVE, model.getSkillLocation(), itemView.getContext());
+                    setRating(true, true);
+                }
+            }
+        });
 
-    @OnClick
-    public void thumbsDownClick() {
-        thumbsDown.setImageResource(R.drawable.thumbs_down_solid);
-        if (!model.isPositiveRated() && !model.isNegativeRated()) {
-            rateSusiSkill(Constant.NEGATIVE, model.getSkillLocation(), itemView.getContext());
-            setRating(true, false);
-        } else if (model.isPositiveRated() && !model.isNegativeRated()) {
-            setRating(false, true);
-            thumbsUp.setImageResource(R.drawable.thumbs_up_outline);
-            rateSusiSkill(Constant.NEGATIVE, model.getSkillLocation(), itemView.getContext());
-            sleep(500);
-            rateSusiSkill(Constant.NEGATIVE, model.getSkillLocation(), itemView.getContext());
-            setRating(true, false);
-        } else if (!model.isPositiveRated() && model.isNegativeRated()) {
-            rateSusiSkill(Constant.POSITIVE, model.getSkillLocation(), itemView.getContext());
-            setRating(false, false);
-            thumbsDown.setImageResource(R.drawable.thumbs_down_outline);
-        }
+        thumbsDown.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View view) {
+                Timber.d(model.isPositiveRated() + " " + model.isNegativeRated());
+                if (!model.isPositiveRated() && !model.isNegativeRated()) {
+                    thumbsDown.setImageResource(R.drawable.thumbs_down_solid);
+                    rateSusiSkill(Constant.NEGATIVE, model.getSkillLocation(), itemView.getContext());
+                    setRating(true, false);
+                }
```

Comment:
This is used everywhere, create a function and reuse

Reply:
@iamareebjamal make a function for this in each file ?

Reply:
No, make a static function

Reply:
@iamareebjamal each viewholder has different view id as different layouts , so where should I place the common function?

Reply:
What's same?

Reply:
actually I don't think there is anything common among the different view holders, but in each viewholder this code is written twice so a method for this can be made in each viewholder

Reply:
OK, let's start with that

Reply:
private void setThumbsUpThumbsDown(boolean rating) {
        if (!model.isPositiveRated() && !model.isNegativeRated()) {
            if (rating) {
                thumbsUp.setImageResource(R.drawable.thumbs_up_solid);
                rateSusiSkill(Constant.POSITIVE, model.getSkillLocation(), itemView.getContext());
                setRating(true, true);
            } else {
                thumbsDown.setImageResource(R.drawable.thumbs_down_solid);
                rateSusiSkill(Constant.NEGATIVE, model.getSkillLocation(), itemView.getContext());
                setRating(true, false);
            }
        }
    }

Reply:
I'm seeing a lot of duplicate code, I don't know how you cannot see it

I'll look at it and let you know what to do